File: Homework 2/Codes/Gozubuyuk_Bulut.rb
puts '-------------pretest'
#pretest

$i = 0
while $i <3 do
  puts $i
  $i = 1 + $i
end

puts '-------------posttest'
#posttest

$i = 0
begin
  puts $i
  $i = 1 + $i
end while $i < 3

puts '-------------break'
#break

$i = 0
while $i <3 do
  if $i ==2
    break
  end
  puts $i
  $i = 1 + $i
end


puts '-------------continue'
#continue


$i = 0
while $i <3 do
  $i = 1 + $i
  if $i ==2
    next
  end
  puts $i
end

File: Homework 2/Codes/Gozubuyuk_Bulut.php
<?php
echo ("----------------pretest\n");
// pretest
$i = 0;
while ($i < 3){
    echo ("$i\n");
    $i++;
}

echo ("----------------posttest\n");
// posttest
$i = 0;
do {
    echo ("$i\n");
    $i++;
} while($i < 3);

echo ("----------------break\n");
// break
$i = 0;
while ($i < 3){
    if ($i == 2) break;
    echo ("$i\n");
    $i++;
}

echo ("----------------continue\n");
// continue
$i = 0;
while ($i < 3){
    $i++;
    if ($i == 2) continue;
    echo ("$i\n");
}

echo ("----------------breakouterloop\n");
// breakouterloop

$i = 0;
$j = 0;
while ($i < 2) {
    while ($j < 2) {
        echo("$i, $j\n");
        $j++;
        if ($i == $j) {break 2;}
    }
    $i++;
    echo("$i\n");
}